File: python/tracing/dify/run_all.py
"""Run the full Dify tracing example set in separate processes."""

from __future__ import annotations

import os
import subprocess
import sys
import uuid
from pathlib import Path

EXAMPLE_DIR = Path(__file__).resolve().parent
SCRIPTS = [
    "01_chat_blocking.py",
    "02_chat_streaming.py",
    "03_completion.py",
    "04_workflow_and_api.py",
    "05_respan_context_and_files.py",
    "06_async_chat_and_workflow.py",
    "07_knowledge_workspace.py",
]


def main() -> None:
    run_id = os.getenv("RESPAN_EXAMPLE_RUN_ID") or f"dify-py-{uuid.uuid4().hex[:12]}"
    child_env = {**os.environ, "RESPAN_EXAMPLE_RUN_ID": run_id}
    print(f"example_run_id={run_id}", flush=True)
    for script in SCRIPTS:
        print(f"\n### Running {script}", flush=True)
        subprocess.run(
            [sys.executable, script],
            cwd=EXAMPLE_DIR,
            env=child_env,
            check=True,
        )


if __name__ == "__main__":
    main()

File: typescript/tracing/dify/run_all.ts
#!/usr/bin/env node
import { spawnSync } from "node:child_process";
import { randomUUID } from "node:crypto";
import path from "node:path";
import { fileURLToPath } from "node:url";

const exampleDir = path.dirname(fileURLToPath(import.meta.url));
const scripts = [
  "01_chat_blocking.ts",
  "02_chat_streaming.ts",
  "03_completion_workflow.ts",
  "04_knowledge_workspace.ts",
  "05_file_and_error.ts",
];
const runId = process.env.RESPAN_EXAMPLE_RUN_ID ?? `dify-ts-${randomUUID().slice(0, 12)}`;
const childEnv = { ...process.env, RESPAN_EXAMPLE_RUN_ID: runId };

console.log(`example_run_id=${runId}`);

for (const script of scripts) {
  console.log(`\n### Running ${script}`);
  const result = spawnSync(process.execPath, ["--import", "tsx", script], {
    cwd: exampleDir,
    env: childEnv,
    stdio: "inherit",
  });
  if (result.error) throw result.error;
  if (result.status !== 0) process.exit(result.status ?? 1);
}
